src/main/java/Sam.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Scanner;
+
+public class Sam {
+    public static void main(String[] args) {
+        // Printing the logo and greeting message
+        String logo = "  _________   _____      _____   \n" +
+                " /   _____/  /  _  \\    /     \\  \n" +
+                " \\_____  \\  /  /_\\  \\  /  \\ /  \\ \n" +
+                " /        \\/    |    \\/    Y    \\\n" +
+                "/_______  /\\____|__  /\\____|__  /\n" +
+                "        \\/         \\/         \\/ ";
+        System.out.println("Hello! I'm SAM\n" + logo + "\n" + "What can I do for you?\n");
+
+        // Initializing scanner to read user input
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+
+        // Array to store tasks
+        Task[] records = new Task[100];
+        int numItems = 0; // Number of tasks
+        int listIndex; // Index for tasks
+
+        // Main loop
+        while (!line.equals("bye")) {
+            String[] words = line.split(" ");
+            printLine();
+            try {
+                // Checking user input
+                switch (words[0]) {
+                case "list":
+                    // Listing tasks
+                    System.out.println("Here are the tasks in your list:");
+                    for (int j = 0; j < numItems; j++) {
+                        System.out.println((j + 1) + "." + records[j]);
+                    }
+                    System.out.println("Now you have " + numItems + (numItems == 1 ? " task " : " tasks ") + "in the list.");
+                    break;
+                case "mark":
+                    // Marking a task as done
+                    listIndex = Integer.parseInt(words[1]) - 1;
+                    markCheck(listIndex, numItems);
+                    records[listIndex].isDone = true;
+                    System.out.println("Nice! I've marked this task as done:\n" + records[listIndex]);
+                    break;
+                case "unmark":
+                    // Marking a task as not done
+                    listIndex = Integer.parseInt(words[1]) - 1;
+                    markCheck(listIndex, numItems);
+                    records[listIndex].isDone = false;
+                    System.out.println("OK, I've marked this task as not done yet:\n" + records[listIndex]);
+                    throw new InvalidInputException();
+                    //break;
```

Comment:
Need to add '\\Fallthrough' comment if there is no break statement inside the case block, as per coding standards
![image](https://github.com/nus-cs2113-AY2324S2/ip/assets/69302316/2e907049-4413-47d2-86dd-086f8fdd6fb2)